src/req-manager.js

```diff
@@ -36,10 +36,10 @@ Req.prototype.load = function (modname) {
         return m.require(modname);
       }
       let relativePath = path.join(path.dirname(self.filepath), modname);
-      if (fs.existsSync(relativePath)) {
-        return m.require(relativePath);
+      if (!fs.existsSync(relativePath)) {
+        return m.require(modname);
```

Comment:
I don't think the use case you have mention in [issue](https://github.com/getgauge/gauge-js/issues/271) works with this fix. Correct me if it is otherwise.
If example is a .js file. `m.require('../../test/example')` will not work it has to be `m.require('../../test/example.js')`

Reply:
@Apoorva-GA : I thought the same way but unfortunately that's not true.
`m.require('../../test/example')` very well get resolved to `example.js`. 
I've tested this piece of code locally and tested in automation tests.

Also, we are using the same imports in our automation cases and that very well works with v2.3.5 just because the `js` files are getting resolved with the same piece of code.

**Recalling code snippet from **v2.3.5****
```
return modname === path.basename(modname) ?  m.require(modname) : m.require(path.join(path.dirname(self.filepath), modname))
```

I believe that by making the below change, we are still giving way to any unknown errors since we have rigidly checked for only `.js` extension.
```
fs.existsSync(relativePath.concat(".js"))
```
Hope this clarifies.

Reply:
Yes. you are right. My test failed locally.